lib/model: Don't mark deleted local only as locally changed (fixes #6432)

imsodin opened the pull request:
> A file that only ever existed locally, but is now deleted, keeps showing up as a locally changed item. That has no value, as for all practical intents and purposes it does not exist.

Like with similar special conditions on send only folder, this adds a stage to pulling that checks for this situation and fixes it by removing the local flag.

calmh commented on lib/db/structs.go:
```diff
@@ -129,19 +129,19 @@ func (f FileInfoTruncated) FileModifiedBy() protocol.ShortID {
 
 // copyToFileInfo just copies all members of FileInfoTruncated to protocol.FileInfo
-func (f FileInfoTruncated) copyToFileInfo() protocol.FileInfo {
+func (f FileInfoTruncated) CopyToFileInfo() protocol.FileInfo {
```
Somewhat skeptical of exposing this method given the latest change to putFile and the fact that the resulting FileInfo can't be easily or safely passed back into the database layer. The Deleted/Ignored cases above are the exceptions where this is safe (as those shouldn't have block lists).

imsodin replied:
I added a new method that converts deleted, truncated file infos to protocol file infos and panics if the file isn't actually deleted.

imsodin commented:
lgtm?

calmh commented:
A test for this situation would be cool...

imsodin commented:
I wrote the test and it passes on master - in hindsight obviously: Deleted files that only we had are handled by scanning (already tested),  and if the remote deletes a file we already deleted we pull that deletion thus removing the locally changed item. I'll file the test separately.

calmh commented:
So test revealed there was no bug, or bug is something other than we though?

imsodin commented:
Initially I thought it was "no bug", but right now I am writing another test as I have suspicion for "https://github.com/syncthing/syncthing/pull/6433#issuecomment-604987278" :)
